### PR TITLE
[Merged by Bors] - feat(CategoryTheory): transport Kan extensions via equivalences

### PR DESCRIPTION
**Pull request by joelriou**
In this PR, it is shown that left/right Kan extensions of functors `F : C ⥤ H` along a functor `L : C ⥤ D` are compatible with changing the functors `F` and `L` by isomorphic functors, by the precomposition with an equivalence `G : C' ⥤ C`, and postcomposition with an equivalence `D ⥤ D'`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by erdOne** [comment]
LGTM, but maybe @dagurtomas can give it a look as well?

**Comment by dagurtomas** [comment]
Thanks for the ping. I'll take a look but I don't have time until tomorrow evening

File: Mathlib/CategoryTheory/Functor/KanExtension/Basic.lean [hunk 1]
```diff
@@ -266,35 +268,145 @@ end
...
-  have := CostructuredArrow.isEquivalence_pre ((whiskeringLeft D D' H).obj e.functor)
-    ((whiskeringLeft C D H).obj L) F
-  exact Functor.asEquivalence (CostructuredArrow.pre ((whiskeringLeft D D' H).obj e.functor)
-    ((whiskeringLeft C D H).obj L) F)
+/-- The functor `LeftExtension L' F ⥤ LeftExtension L F` induced by an isomorphism
+`L ⋙ G ≅ L'`. -/
+@[simps!]
+def LeftExtension.postcomp₁ : LeftExtension L' F ⥤ LeftExtension L F :=
+  StructuredArrow.map₂ (F := (whiskeringLeft D D' H).obj G) (G := 𝟭 _) (𝟙 _)
+    ((whiskeringLeft C D' H).map e.inv)
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Functor/KanExtension/Basic.lean:278]: Here we don't need an isomorphism of functors, and `e.inv` could be replaced by a natural transformation `L' ⟶ L ⋙ G`.

**Review by dagurtomas — APPROVED** [review]
I just suggested a minor generalisation, otherwise LGTM.

File: Mathlib/CategoryTheory/Functor/KanExtension/Basic.lean (hunk 1 above)
Comment (joelriou) [reply Mathlib/CategoryTheory/Functor/KanExtension/Basic.lean]: Thanks for the suggestion, which I have implemented.

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Comment by riccardobrasca** [comment]
This is great, thanks!

bors merge

**Pull request closed without merging**